roles/lib_utils/library/oo_azure_rm_publish_image.py

```diff
@@ -194,19 +228,28 @@ def run_ansible(params):
 
         if params['state'] == 'offer':
             results = apc.manage_offer(params)
+        elif params['state'] == 'publish':
+            results = apc.publish(offer=params['offer'], emails=params['emails'])
         elif params['state'] == 'cancel_op':
-            results = apc.cancel_operation(offer=params['offer'])
+            results = apc.cancel_and_wait_for_operation(params)
         elif params['state'] == 'go_live':
             results = apc.go_live(offer=params['offer'])
         else:
-            raise AzurePublisherException('Unsupported query type: {}'.format(params['query']))
+            raise AzurePublisherException('Unsupported query type: {}'.format(params['state']))
 
         changed = False
 
-        if results.status_code in [200, 201]:
+        if results.status_code in [200, 201, 202]:
             changed = True
 
-        return {'data': results.json(), 'changed': changed, 'status_code': results.status_code}
+        rval = None
```

Comment:
under what circumstances is this change necessary, out of interest?

Reply:
When calling `cancel_operation` there is no data returned in the results.  When calling `results.json()` an exception is thrown since there is nothing returned.  This exception handling allows us to handle emtpy json responses as long as the `results.ok` is `True`.